tests/hikari/models/test_users.py

```diff
@@ -33,6 +38,123 @@ def test_PremiumType_str_operator():
     assert str(type) == "NITRO_CLASSIC"
 
 
-def test_PartialUser_str_operator():
-    mock_user = mock.Mock(users.PartialUserImpl, username="thomm.o", discriminator="8637")
-    assert users.PartialUserImpl.__str__(mock_user) == "thomm.o#8637"
+class TestUser:
+    @pytest.fixture
+    def obj(self):
+        class StubUser(users.User):
+            app = None
+            id = 123
+            discriminator = "0001"
+            username = None
+            avatar_hash = None
+            is_bot = None
+            is_system = None
+            flags = None
+            mention = None
+
+        return hikari_test_helpers.unslot_class(StubUser)()
```

Comment:
is unslot the safest thing to use here? if we want to change class methods specifically couldn't we use the mock_class_namespace helper without discarding the slots and losing the safety those provide?

Reply:
The thing is that I don't want the functions I mock sometimes to be mocked. I just want to mock then when I need to, thus why I unslot

Reply:
Is it not possible to mock the name space for tests around properties that specifically need it, I'm not really a fan of the idea of in slotting objects everywhere for our tests because of the implications this has

Reply:
Mocking the namespace later doesn't let me set specific attributes in each test

Reply:
Is that not more an issue of using an unrealistic class where variables are set on the class definition rather than as instance variables because slotting will make those read only, is it not possible to use a more realistic model that's possibly attrs based?

Reply:
The idea of using an instance of the actual object is that if, for example, someone opens a pr that changes where `avatar` is located to `UserImpl`, and we are using `UserImpl` to test this, then the tests won't fail, but they should.

Doing it this was just assures less cases when this could have a false positive.

Reply:
Well this is no longer an instance of the actual object, this is an instance of an object that inherits from it and overwrites all the abstractmethod with class attributes while throwing away the slotting which would normally make class attributes read only on objects for us. Not sure how that compares to making a stub attrs model where you'd at least get proper slotting.

Reply:
Solved